blotztask-ui/src/app/dashboard/today/page.tsx

```diff
@@ -50,17 +54,29 @@ export default function Today() {
     }
   };
 
-  const handleAddTask = (taskTitle) => {
-    console.log('Adding task:', taskTitle);
-    // Implement add task logic here , going to api to add task
+  const handleAddTask = async (newTaskData: AddTaskItemDTO) => {
+    console.log('Adding task:', newTaskData);
+    await submitNewTask(newTaskData);
+    setReload((prev) => !prev);
+  };
+
+  const submitNewTask = async (newTaskData: AddTaskItemDTO) => {
+    try {
+      console.log('Before adding task:', newTaskData);
+      await addTaskItem(newTaskData);
+      console.log('Task added successfully! Now loading tasks...');
```

Comment:
```suggestion
loadTasks()
```
call the loadTask here should reload the task list already

Reply:
Okay I can do that